homeassistant/components/stream/__init__.py

```diff
@@ -225,7 +226,7 @@ async def async_handle_record_service(hass, call):
     # Take advantage of lookback
     hls = stream.outputs.get("hls")
     if lookback > 0 and hls:
-        num_segments = min(int(lookback // hls.target_duration), hls.num_segments)
+        num_segments = min(int(lookback // hls.target_duration), MAX_SEGMENTS)
```

Comment:
Could this be the problem? If there are no segments ready yet, hls.target_duration gives a 0 and this will blow up

Reply:
That's not the issue. I am getting segments back, but when the recorder worker opens them, the pts/dts of the packets start at 0 instead of `segment.start_pts[0]` like they do with every other segment after that.

Reply:
I really can't see where your changes affect that though, unless my hijacking of the `hls` plugin for use in `recorder` is grabbing segments after they've already been modified by the `m4s` helpers.